overlord/snapstate/snapstate.go

```diff
@@ -1266,6 +1266,11 @@ func InstallPathWithDeviceContext(st *state.State, si *snap.SideInfo, path, name
 	opts *RevisionOptions, userID int, flags Flags, prqt PrereqTracker,
 	deviceCtx DeviceContext, fromChange string) (*state.TaskSet, error) {
 	logger.Debugf("installing from local file with device context %s", name)
+
+	if !opts.Revision.Unset() && si.Revision != opts.Revision {
+		return nil, fmt.Errorf("cannot install local snap with mismatch required revision: %v != %v", opts.Revision, si.Revision)
```

Comment:
Super nitpick, and maybe you have a reason for not doing this
```suggestion
		return nil, fmt.Errorf("cannot install local snap %q: %v != %v (revision mismatch)", name, opts.Revision, si.Revision)
```